packages/@glimmer/integration-tests/lib/modes/rehydration/builder.ts

```diff
@@ -12,7 +12,7 @@ export class DebugRehydrationBuilder extends RehydrateBuilder {
     if (node.nodeType !== 8) {
       if (el.nodeType === 1) {
         // don't stat serialized cursor positions
-        if (el.tagName !== 'SCRIPT' && !el.getAttribute('gmlr')) {
+        if (el.tagName !== 'SCRIPT' || !el.getAttribute('glmr')) {
```

Comment:
Doesn’t this need to be `&&` still? If I’m reading this correctly, this will now match *any* script element, not just those with the `glmr` attribute. (Great catch on the typo!)

Reply:
I think it needs to be `||`. Please correct me if I'm wrong, but I believe that the point of this code is to track the removal of nodes that are NOT an element like the remote element marker. If I'm wrong, the following is totally wrong...

If I'm right, then this code can be written more like:
```
if (!(el.tagName === 'SCRIPT' && el.getAttribute('glmr'))) { // if not a remote element marker
  this.clearNodes.push(node);
}
```
By [De Morgan's Laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws) (reaching back into my digital design days), these two are equivalent: 
```
const A = el.tagName === 'SCRIPT';
const B = el.getAttribute('glmr');
!(A && B) === (!A || !B) // these are equivalent
```
If you want me to switch to the other form, I'm totally in favor of it as I believe it's probably more readable but I thought the original author was intending to apply the transformation.

Reply:
This is what I get for doing code reviews past my bedtime… thanks for the thorough explanation!